### DISCUSSION TITLE
4000+ characters long access Token after Login

### DISCUSSION THREAD

**Discussion by NickRTR** [supabase/supabase#8895] [Questions] [answered]
This issue occured suddenly, on a codebase that has been working without issues over a long time. Therefore I thought the issue might come from Supabase.

**To clarify:**

When logging in, the reponse object from supabase is as usually with the twist that the access token in data and session is more than 4000 characters long. Hence, I'm not able to set a cookie, because a cookie's max length is 4096.

Part of Response access_token:
![image](https://user-images.githubusercontent.com/95316598/189493098-e96b7e76-d744-454e-aac7-2d9d504f2095.png)

If someone can help me out, it would be highly appreciated.

**Comment by GaryAustin1** [comment, marked as answer]
Run it thru jwt.io and see if it is valid.  Maybe you have some long structure in raw_app_meta_data or raw_user_meta_data in the auth.users table.

Is it for a particular user?

**Reply by NickRTR** [reply]
Oh, I didn't know the user metadata gets send with the JWT token. I have two big objects in every users metadata, yes.

**Reply by NickRTR** [reply]
I thought I could store it in the users metadata without any drawbacks. And I haven't seen any warning in the docs.

**Reply by GaryAustin1** [reply]
You might want to consider storing it in a profile table.  
There is this mention of removing it from the jwt:  https://github.com/supabase/gotrue/pull/415  ,but no time frame on when it might be turned on.

**Reply by NickRTR** [reply]
I had actually been storing my user data in a seperate database until around one month ago. I guess I'll have to rework this part of my application again. Thanks for your advice and have a great day!
